pkg/utils/helper.go

```diff
@@ -155,11 +162,18 @@ func (s *SecurityGroupForPods) filterPodSecurityGroups(
 			continue
 		}
 
+		sgNameList = append(sgNameList, sgp.Spec.SecurityGroupNames.GroupNames...)
 		sgList = append(sgList, sgp.Spec.SecurityGroups.Groups...)
 	}
 
+	sgNameList = RemoveDuplicatedSg(sgNameList)
```

Comment:
It would be preferable to enforce this via CRD validation in `apis/vpcresources/v1beta1/securitygrouppolicy_types.go` for both SG IDs and names, like `+kubebuilder:validation:UniqueItems=true`.

Reply:
make sense let me do that. It does make it slightly harder to validate just at the CRD if we are mixing SG id and SG names though